addons/crm/models/crm_lead.py

```diff
@@ -1197,26 +1197,52 @@ def _get_rainbowman_message(self):
             AND
                 probability = 100
             AND
-                DATE_TRUNC('year', date_closed) = DATE_TRUNC('year', CURRENT_DATE)
+                DATE_TRUNC('year', date_closed) = DATE_TRUNC('year', %(tz_midnight)s)
             AND
                 (user_id = %(user_id)s OR team_id = %(team_id)s)
         """
-        self.env.cr.execute(query, {'user_id': self.user_id.id,
-                                    'team_id': self.team_id.id})
+        tz_midnight = fields.Datetime.now().astimezone(pytz.timezone(self.user_id.tz or 'UTC')).replace(hour=0, minute=0, second=0)
```

Comment:
Unsure about the computation... I suppose the idea is: take now in the current user's timezone, then take the beginning of the day. And not sure we should take the responsible timezone... in the end, the rainbowman is displayed for the current user, not the assigned user (probably often the same but hey).

Hence: why not simply
```suggestion
        day_begin_user_tz = datetime.combine(
            fields.Date.today(),
            time.min,
        )
```
and using it directly in the query ?

Reply:
I think it makes more sense to use the responsible ID, given that this is what will be matched against the leads to determine five-day combos, streaks, and records. If we change this for timezones, would it make sense to also change the user_id that's checked against the responsible database ID to determine records?

(also, `day_begin_user_tz` here isn't midnight in the current user's timezone, it is midnight in UTC today without a timezone. Using context_today to calculate it also would not work because it is a date without time info, although I guess you could apply the UTC time it would be at midnight at this date, but I don't think it would simplify the code much?)

Reply:
Jean Luc moving the win on Micheline's lead will record the win towards Micheline's statistics regardless, no?

I'll change it :)